panda.py

```diff
@@ -0,0 +1,224 @@
+from pxr import Usd, Vt
+import json
+
+data = {}
+stage = Usd.Stage.Open('franka.usda')
+panda = stage.GetPrimAtPath('/panda')
+
+link0 = stage.GetPrimAtPath('/panda/panda_link0')
+joint1 = stage.GetPrimAtPath('/panda/panda_link0/panda_joint1')
+
+link1 = stage.GetPrimAtPath('/panda/panda_link1')
+joint2 = stage.GetPrimAtPath('/panda/panda_link1/panda_joint2')
+
+link2 = stage.GetPrimAtPath('/panda/panda_link2')
+joint3 = stage.GetPrimAtPath('/panda/panda_link2/panda_joint3')
+
+link3 = stage.GetPrimAtPath('/panda/panda_link3')
+joint4 = stage.GetPrimAtPath('/panda/panda_link3/panda_joint4')
+
+link4 = stage.GetPrimAtPath('/panda/panda_link4')
+joint5 = stage.GetPrimAtPath('/panda/panda_link4/panda_joint5')
+
+link5 = stage.GetPrimAtPath('/panda/panda_link5')
+joint6 = stage.GetPrimAtPath('/panda/panda_link5/panda_joint6')
+
+link6 = stage.GetPrimAtPath('/panda/panda_link6')
+joint7 = stage.GetPrimAtPath('/panda/panda_link6/panda_joint7')
+
+link7 = stage.GetPrimAtPath('/panda/panda_link7')
+joint8 = stage.GetPrimAtPath('/panda/panda_link7/panda_joint8')
+
+link8 = stage.GetPrimAtPath('/panda/panda_link8')
+hand_joint = stage.GetPrimAtPath('/panda/panda_link8/panda_hand_joint')
+
+hand = stage.GetPrimAtPath('/panda/panda_hand')
+finger_joint1 = stage.GetPrimAtPath('/panda/panda_hand/panda_finger_joint1')
+finger_joint2 = stage.GetPrimAtPath('/panda/panda_hand/panda_finger_joint2')
+
+print(joint1.GetPropertyNames())
+
+#Joint Limits
+joint1_uL = joint1.GetAttribute('physics:upperLimit').Get()
+joint1_lL = joint1.GetAttribute('physics:lowerLimit').Get()
+joint2_uL = joint2.GetAttribute('physics:upperLimit').Get()
+joint2_lL = joint2.GetAttribute('physics:lowerLimit').Get()
+joint3_uL = joint3.GetAttribute('physics:upperLimit').Get()
+joint3_lL = joint3.GetAttribute('physics:lowerLimit').Get()
+joint4_uL = joint4.GetAttribute('physics:upperLimit').Get()
+joint4_lL = joint4.GetAttribute('physics:lowerLimit').Get()
+joint5_uL = joint5.GetAttribute('physics:upperLimit').Get()
+joint5_lL = joint5.GetAttribute('physics:lowerLimit').Get()
+joint6_uL = joint6.GetAttribute('physics:upperLimit').Get()
+joint6_lL = joint6.GetAttribute('physics:lowerLimit').Get()
+joint7_uL = joint7.GetAttribute('physics:upperLimit').Get()
+joint7_lL = joint7.GetAttribute('physics:lowerLimit').Get()
+# joint8_uL = joint8.GetAttribute('physics:upperLimit').Get()
+# joint8_lL = joint8.GetAttribute('physics:lowerLimit').Get()
+
+
+data['limits'] = []
+data['limits'].append([[joint1_lL, joint1_uL],
+                      [joint2_lL, joint2_uL],
+                      [joint3_lL, joint3_uL],
+                      [joint4_lL, joint4_uL],
+                      [joint5_lL, joint5_uL],
+                      [joint6_lL, joint6_uL],
+                      [joint7_lL, joint7_uL]])
+                    #   [joint8_lL, joint8_uL]])
+
+
+#DOF
+
+data['num_dof'] = 7
```

Comment:
Since we know the num_dof for the panda, we may wish to use this in the beginning to seed our loops!

Can we extract this from the USD file without much prior knowledge?